experimental/net/net.go

```diff
@@ -0,0 +1,90 @@
+package net
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+)
+
+// NetConfig is a configuration for TCPListeners.
+type NetConfig interface {
```

Comment:
I completely misunderstood you were suggesting to put some `Config` _in_ the context key. I just went straight to (more or less) copying what was being done for Function Listeners! I suppose I have read Context + Listeners and I jumped straight to that conclusion!

Reply:
but I'll need `NewNetConfig()` in `experimental/net`, otherwise it won't be public. And if I need that to be there, then I'll need the impl here too to `return &netConfig{}`. Have I misunderstood again? 😅 

`netConfig` and the context key can all be unexported though, because now you would set them through the public API (I'll probably need a public way to extract it from the context though, unless I export the key)

Reply:
if NewNetConfig returns an interface, it shouldn't matter if its implementation is in internal/net